src/Sarif/ExtensionMethods.cs

```diff
@@ -183,45 +183,27 @@ public static string GetMessageText(this Result result, IRule rule, bool concise
             {
                 text = string.Empty;    // Ensure that it's not null.
 
-                string messageId = result.Message?.MessageId;
-
-                if (rule != null && !string.IsNullOrWhiteSpace(messageId))
+                if (rule != null)
                 {
-                    string messageString;
-
-                    string[] arguments = null;
-
-                    if (result.Message?.Arguments != null)
-                    {
-                        arguments = new string[result.Message.Arguments.Count];
-                        result.Message.Arguments.CopyTo(arguments, 0);
-                    }
-                    else
-                    {
-                        arguments = new string[0];
-                    }
+                    string messageId = result.Message?.MessageId;
 
-                    if (rule.MessageStrings?.ContainsKey(messageId) == true)
+                    if (!string.IsNullOrWhiteSpace(messageId)
+                        && rule.MessageStrings?.ContainsKey(messageId) == true)
```

Comment:
>ContainsKey [](start = 48, length = 11)

One last tweak. You can save the double dictionary lookup by using `rule.MessageStrings?.TryGetValue(messageId,out formatString) == true`